### DISCUSSION TITLE
Scan BitBucket private repository using Trivy

### DISCUSSION THREAD

**Discussion by psg18dhc** [aquasecurity/trivy#6832] [Ideas]
### Description

I noticed that when using BitBucket private repositories it's not possible to scan my code repo as i get auth errors.

GITHUB_TOKEN and GITLAB_TOKEN env vars do not work (because it's not a GitHub repo) 

Is there a way to do this securely without having to make the repo public ?

i.e can we have a BITBUCKET_TOKEN env var specifically for this purpose ?

Regards Daniel C

### Target

Git Repository

### Scanner

None

**Comment by nikpivkin** [comment]
Hi @psg18dhc !

At the moment it is only possible to pass token for GitHub and GitLab.  https://aquasecurity.github.io/trivy/v0.51/docs/target/repository/#scanning-private-repositories

@knqyf263 How about adding a flag to pass credentials to different Git hosting platforms?

**Reply by psg18dhc** [reply]
Yes please Teppei. :)

**Comment by nikpivkin** [comment]
Track https://github.com/aquasecurity/trivy/issues/6833

**Comment by FranAguiar** [comment]
Hello, any workaround to scan bitbucket private repositories?

**Reply by FranAguiar** [reply]
Actually, there is a workaround:
`trivy repo https://<user>:<pass>@bitbucket.org/<company>/<repository> --branch main
`

I hope it helps
